src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp

```diff
@@ -89,9 +97,10 @@ class TPackageScanner final : public AbstractHandler<std::shared_ptr<TScanContex
         const auto packageName = std::string(data->packageName().data());
         const auto packageVendor = std::string(data->packageVendor().data());
 
+        const auto cacheKey = osPlatform + "_" + packageVendor + "_" + packageName;
+
         // Search in Level 1 cache
-        if (const auto cacheKey = osPlatform + "_" + packageVendor + "_" + packageName;
-            m_translationL1Cache.isHit(cacheKey))
+        if (m_translationFilter.count(cacheKey) == 0 && m_translationL1Cache.isHit(cacheKey))
```

Comment:
Is this comparison ok?

If we do not have any translation (==0) it will search in L2 in vain.

Reply:
The first attempt always will fall in a cache miss, because a prior we can't know if there is or there isn't a translation for a package. 

The next time, the filter will check first if is a miss, or follow the current flow of miss/hit for L1 and L2.